server/src/main/java/io/pravega/schemaregistry/server/rest/auth/AuthHandlerManager.java

```diff
@@ -43,7 +42,7 @@ private void loadHandlers() {
             if (serverConfig.isAuthEnabled()) {
                 ServiceLoader<AuthHandler> loader = ServiceLoader.load(AuthHandler.class);
                 for (AuthHandler handler : loader) {
-                    if (handler instanceof PasswordAuthHandler) {
+                    if (handler instanceof PasswordAuthHandler && !(handler instanceof BasicAuthHandler)) {
```

Comment:
The intent is not clear here (although its clear that the condition is only looking for `PasswordAuthHandler` objects that aren't of type `BasicAuthHandler`). A comment'd help.

Reply:
This is because presently schema registry service takes a library dependency on pravega controller. this, among others, is to make use of authentication utility classes in controller module. The `PasswordAuthHandler` is the basic auth handler implementation that we want to make use of. 

However, the password auth handler requires GrpcServiceConfig to be passed. So we create a derived class called BasicAuthHandler which takes schema regsitry service config and converts it to grpc service config. 

This part of the code will be replaced once auth utility classes are refactored out of controller module into a separate shared library.

Reply:
Sounds good.